src/app/digital/public/api/impl/DigitalComponent.ts

```diff
@@ -16,4 +18,34 @@ export class DigitalComponentImpl extends ComponentImpl<
     public override get info(): DigitalComponentInfo {
         return new DigitalComponentInfoImpl(this.circuit, this.kind);
     }
+
+    public override firstAvailable(portGroup: Port["group"]): DigitalPort | undefined {
+        const ports = this.internal.getPortsForComponent(this.id);
+
+        const inputTypes = this.info.inputPortGroups;
+        const outputTypes = this.info.outputPortGroups;
+
+        const isInputType = inputTypes.includes(portGroup);
+        const isOutputType = outputTypes.includes(portGroup)
+
+        if (!isInputType && !isOutputType) 
+            return undefined;
```

Comment:
A valid portGroup will always be an input or output group (or we have a bigger problem)
So you can simplify this by just doing
```ts
if (!this.info.portGroups.includes(portGroup))
    return undefined; // Invalid port group for the component
```